LaTeX table writer formatting inconsistent with intended format bugfix and fixing reference formatting

cody-d10 opened the pull request:
Correcting LaTeX table formatting and fixing reference formatting

gsgall committed:
Addressing issue #1 and #11

Updated all of the Makfiles to use rpath rather than set the LD an DYLD_LIBRARY_PATH variables

- explicitly set the linker to be used as well should be using lld
  - need to add this requirement to the conda packages
- modifying the conda packages to also explicitly set the path to the compiler
- added a new environment variable LINKER to set the name of the linker

Initial conda package with compilers added

- Compiler and linkers tools with pinned versions have been added
- At some point I should try and remove as many dependencies that are pinned as possible
- Testing on a linux machine is still required for this

Removing scripts to set compiler flags from activate.d and deactivate.d

- conda already sets the variable CXX to the right compiler and we do not need to
- Setting this variable to the wrong value was causing build issues with the linking step

Removed linker specifier

- At one point I was specifying -fuse-ld in the Makefile but this should not be used if all the compilers are pointing to the correct directories

cody-d10 committed:
Update DefaultTableWriter.C, addressing #7 and #16

cody-d10 committed:
Update reference formatting

gsgall committed:
Updating workflows for testing and website generation

- To ensure the conda packages are in a working state the tester will build and install the package on each run
- This local installation also requires the packages be built in a folder within the project
  - when trying to just use --use-local with the install a permission error arises

- Currently the website generation is not working because of a MOOSE change
  - the versioner capabilities of MOOSE doc are breaking the webite build

gsgall commented on src/Reaction.C:
```diff
@@ -680,9 +680,15 @@ Reaction::operator!=(const Reaction & other) const
-  string temp_refs = "";
-  for (auto r : _references)
-    temp_refs += "\\cite{" + r + "}";
+  string temp_refs = "\\cite{";
+  for (size_t r = 0; r < _references.size(); r++) {
+    temp_refs += _references[r];
+	if (r != _references.size() - 1)
+		temp_refs += ",";
+  }
+  temp_refs += "}";
+  if (temp_refs == "\\cite{}")
+	  temp_refs = "";
```
```suggestion
  string temp_refs = "\\cite{";
  for (size_t i = 0; i < _references.size(); i++) {
    temp_refs += _references[i];
	if (i != _references.size() - 1)
		temp_refs += ", ";
  }
  temp_refs += "}";
```

Since we are not iterating over the objects in the _references vector directly now the loop index should be `i` just for the sake of consistency. Additionally, we will never have a Reaction that has no references since we force all Reactions to have a reference so there is no need for the check at the bottom of the loop.

gsgall commented:
I would also like for you to create a test for these two changes to ensure that we maintain this functionality going forward. If you need additional guidance on creating a test for this stuff please let me know. 

And please be sure to run clang-format on your code as well

https://ncsu-comps-group.github.io/prism/getting_started/contributing.html

gsgall committed:
Some files are inconsistently formatted this applies formatting rules to the project

gsgall committed:
Removing linker requirement from conda package and cleaning up comments in workflows

gsgall committed:
Updating workflows to account for conflict in base environment and moving towards proper conda environments

- A PR in my personal fork will update this package and this is the workflow for the changes in that branch
- This will almost ceratinly break the current workflows
- Needed to add environment activation into each run to ensure it is working properly

gsgall committed:
Removing setting environment variable in order to move towards proper conda managment

- A PR in my personal fork will update this package and this is the workflow for the changes in that branch
- This will almost ceratinly break the current workflows

gsgall committed:
Trying to debug gtest issue

gsgall committed:
trying to debug gtest issue

gsgall committed:
Updating workflows

gsgall committed:
Updating workflows so that they have to be activated

gsgall committed:
Making the main file more usable addressing #13

- added optional flags to allow users to specify files both for input and output
- if no arguments are provided then the example file will run and create the files in the example directory

gsgall committed:
Adding logic so that notes are printed properly addressing #6

- Notes were previously only added on the first appearance of the note
  - the writer has been updated to ensure that notes are added correctly
- Also added files to .gitignore which are produced during latex compilation

gsgall committed:
Updating

gsgall committed:
Updating testing in order to ensure notes functionality does not break

gsgall committed:
Redirecting std in/out

- Adding the configuration in the test case ensures that nothing is
  printed as a result of reading the files during testing

gsgall committed:
Adding input and gold file and updating testing

- Needed to update the tests to dump the redirected output when a test
  has a failure so we can debug easier
- Forogto to add the files for input and the gold for the test

gsgall committed:
Updating the dll testing Makefile for testing

- The Makefile-with-dll did not create the outputs directory so when
  that test was run it would fail since it was unable to open a file in
  that directory

gsgall commented:
You can use [DefaultTableWriterTest](https://github.com/NCSU-ComPS-Group/prism/blob/devel/test/tests/DefaultTableWriterTest.C) as a template for getting started on a test for this functionality.

cody-d10 committed:
Update src/Reaction.C

Co-authored-by: Grayson Gall <66559200+gsgall@users.noreply.github.com>

cody-d10 committed:
Adding test for multiple references

cody-d10 committed:
fixing LaTeX column formatting

gsgall commented:
Looks like it is good to go @cody-d10 and you're passing the tests we have in place. One last thing I would like to see is if you can add the issue numbers which this addresses to your first commit message. To do this you can run 

`git rebase -i HEAD~7` 

go into the file that gets opened change `pick` to `edit` or `e` for your first commit which should have the hash [5a3f1ae](https://github.com/NCSU-ComPS-Group/prism/pull/9/commits/5a3f1ae17e558a0a4a35f3a22bcfafa27573d868). Then you can use the 

`git commit --amend`

And then add to your commit message something along the lines of "addressing #9 and #16"

after that you should be able to finish the rebase with 

`git rebase --continue` 

and you'll probably need to force push for to remote with 

`git push -f`

This should only affect the commit message and help us have a clear history of why all of these changes were made if we ever need to look back at the history.

cody-d10 committed:
Merge branch 'devel' into devel
